webui/src/pages/extension-detail/extension-detail-overview.tsx

```diff
@@ -163,11 +163,47 @@ export const ExtensionDetailOverview: FunctionComponent<ExtensionDetailOverviewP
         </>;
     };
 
-    const renderWorksWithList = (downloads: { [targetPlatform: string]: UrlString }): ReactNode => {
-        return Object.keys(downloads).map((targetPlatform, index) => {
-            const displayName = getTargetPlatformDisplayName(targetPlatform);
-            return displayName ? <span key={targetPlatform}>{index > 0 ? ', ' : ''}{displayName}</span> : null;
-        });
+    const renderWorksWithEngines = (engines?: Record<string, string>): ReactNode => {
+        if (engines == null) {
+            return null;
+        }
+
+        const data = Object.keys(engines)
+            .map((engine) => ({
+                key: engine,
+                name: getEngineDisplayName(engine),
+                version: engines[engine]
```

Comment:
I wonder how much free rein extensions have here. Do we have any limits on what users can put into the engine version?
I don't think it's necessary to parse these values per se, but we could have some validation on the backend if we don't already.

Reply:
The `vscode` version is required, but a publisher can add other engines too. From https://code.visualstudio.com/api/references/extension-manifest:
![image](https://github.com/user-attachments/assets/73f7d785-05bf-41ea-922d-a8f8f1b45d51)